apps/crn-server/src/handlers/webhooks/webhook-contentful.ts

```diff
@@ -0,0 +1,86 @@
+import 'source-map-support/register';
+import Boom from '@hapi/boom';
+import { EventBridge } from 'aws-sdk';
+import { APIGatewayEvent, Handler } from 'aws-lambda';
+import { WebhookDetailType } from '@asap-hub/model';
+import { framework as lambda } from '@asap-hub/services-common';
+import { ContentfulWebhookPayload } from '@asap-hub/contentful';
+import { sentryWrapper } from '../../utils/sentry-wrapper';
+import {
+  contentfulWebhookAuthenticationToken,
+  eventBridgeAccessKey,
+  eventBridgeEndpoint,
+  eventBridgeSecret,
+  eventBus,
+  eventSource,
+} from '../../config';
+
+export const contentfulWebhookFactory = (
+  webhookAuthenticationToken: string,
+  eventBridge: EventBridge,
+): lambda.Handler =>
+  lambda.http<Record<string, string>, void>(
+    async (request: lambda.Request<Record<string, string>>) => {
+      if (!validateContentfulRequest(request, webhookAuthenticationToken)) {
+        throw Boom.unauthorized();
+      }
+
+      const detailType = getDetailTypeFromRequest(request);
+
+      await eventBridge
+        .putEvents({
+          Entries: [
+            {
+              EventBusName: eventBus,
+              Source: eventSource,
+              DetailType: detailType,
+              Detail: JSON.stringify(request.payload),
+            },
+          ],
+        })
+        .promise();
+
+      return {
+        statusCode: 200,
+      };
+    },
+  );
+
+const eventBridge = new EventBridge({
+  endpoint: eventBridgeEndpoint,
+  accessKeyId: eventBridgeAccessKey,
+  secretAccessKey: eventBridgeSecret,
+});
+
+export const handler: Handler<APIGatewayEvent> = sentryWrapper(
+  contentfulWebhookFactory(contentfulWebhookAuthenticationToken, eventBridge),
+);
+
+export const validateContentfulRequest = (
+  request: lambda.Request,
+  webhookAuthenticationToken: string,
+): request is lambda.Request<ContentfulWebhookPayload<'teams' | 'news'>> => {
+  if (!request.headers.authorization) {
+    return false;
+  }
+
+  if (request.headers.authorization !== webhookAuthenticationToken) {
+    return false;
+  }
+
+  return true;
+};
+
+const getDetailTypeFromRequest = (
+  request: lambda.Request<ContentfulWebhookPayload<'teams' | 'news'>>,
+): WebhookDetailType => {
+  const actions = request.headers['x-contentful-topic']?.split('.') ?? [];
+  const action = actions[actions.length - 1] as 'publish' | 'unpublish';
```

Comment:
Do we want to handle only `publish` and `unpublish`?

I saw here https://www.contentful.com/developers/docs/references/content-management-api/#/reference/webhooks that this `x-contentful-topic` can have these other values
<img width="694" alt="Screenshot 2023-03-30 at 14 39 46" src="https://user-images.githubusercontent.com/16595804/228919554-092b9289-0e2a-44f9-a827-a1deb57019ef.png">

Reply:
I think because contentful allows creating/editing _without_ publishing then these are probably the only events we'll need. To archive or delete you need to have previously unpublished.

Reply:
One possible concern that might be worth checking though is that if we unpublish some content, the management API will (potentially) still return it, so the code that deletes from algolia when a 404 is returned from the `fetchById` might need to be double checked that it works correctly to delete when content is unpublished.

Reply:
i should have probably added it to this PR's description - for our purposes we are only interested in publish and `unpublish` events. Any delete/archive action is, like @lennym said, preceded by an `unpublish` event